Test/Unit/Model/CatalogIngestion/ProductEventManagerTest.php

```diff
@@ -0,0 +1,315 @@
+    }
+
+    /**
+     * @test
+     */
+    public function testRunInstantProductEvent_withoutAsync()
+    {
+        $store = $this->objectManager->get(StoreManagerInterface::class);
+        $websiteId = $store->getWebsite()->getId();
+        $configData = [
+            [
+                'path' => BoltConfig::XML_PATH_CATALOG_INGESTION_INSTANT_ENABLED,
+                'value' => 1,
+                'scope' => ScopeInterface::SCOPE_WEBSITES,
+                'scopeId' => $websiteId,
+            ],
+            [
+                'path' => BoltConfig::XML_PATH_CATALOG_INGESTION_INSTANT_ASYNC_ENABLED,
+                'value' => 0,
+                'scope' => ScopeInterface::SCOPE_WEBSITES,
+                'scopeId' => $websiteId,
+            ]
+        ];
+        TestUtils::setupBoltConfig($configData);
+
+        $apiHelper = $this->createPartialMock(ApiHelper::class, ['sendRequest']);
+        $apiHelper->expects(self::once())->method('sendRequest')->willReturn(true);
+        TestHelper::setProperty($this->productEventManager, 'apiHelper', $apiHelper);
+        $product = $this->createProduct();
+
+        $this->productEventManager->runInstantProductEvent(
+            $product->getId(),
+            ProductEventInterface::TYPE_UPDATE
+        );
+    }
+
+    /**
+     * @test
+     */
+    public function testRunInstantProductEvent_withAsync()
+    {
+        $amqpConfigExist = false;
+        if (class_exists('Magento\Framework\Amqp\Config')) {
+            $queueConfig = $this->deploymentConfig->getConfigData(AmqpConfig::QUEUE_CONFIG);
+            $amqpConfigExist = isset($queueConfig[AmqpConfig::AMQP_CONFIG])? true : false;
+        }
+
+        if ($this->moduleManger->isEnabled('Magento_AsynchronousOperations') && !$amqpConfigExist) {
+            $this->expectExceptionMessage('Unknown connection name amqp');
+        }
+
+        if (!$this->moduleManger->isEnabled('Magento_AsynchronousOperations') && !$amqpConfigExist) {
+            $this->expectExceptionMessage('Magento Asynchronous Operations is not supported on your magento version, please verify.');
+        }
+
+        $store = $this->objectManager->get(StoreManagerInterface::class);
+        $websiteId = $store->getWebsite()->getId();
+        $configData = [
+            [
+                'path' => BoltConfig::XML_PATH_CATALOG_INGESTION_INSTANT_ENABLED,
+                'value' => 1,
+                'scope' => ScopeInterface::SCOPE_WEBSITES,
+                'scopeId' => $websiteId,
+            ],
+            [
+                'path' => BoltConfig::XML_PATH_CATALOG_INGESTION_INSTANT_ASYNC_ENABLED,
+                'value' => 1,
+                'scope' => ScopeInterface::SCOPE_WEBSITES,
+                'scopeId' => $websiteId,
+            ]
+        ];
+        TestUtils::setupBoltConfig($configData);
+
+        $product = $this->createProduct();
+
+        $this->productEventManager->runInstantProductEvent(
```

Comment:
Could we test the internal objects that created during this call?

Reply:
Added rabbitmq queue assert cases, after creation